runtime/protocol/http-client-engines/http-client-engine-okhttp/jvm/test/aws/smithy/kotlin/runtime/http/engine/okhttp/CustomCertificateTest.kt

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package aws.smithy.kotlin.runtime.http.engine.okhttp
+
+import aws.smithy.kotlin.runtime.content.decodeToString
+import aws.smithy.kotlin.runtime.http.HttpBody
+import aws.smithy.kotlin.runtime.http.HttpMethod
+import aws.smithy.kotlin.runtime.http.SdkHttpClient
+import aws.smithy.kotlin.runtime.http.request.HttpRequest
+import aws.smithy.kotlin.runtime.http.request.url
+import aws.smithy.kotlin.runtime.http.toByteStream
+import aws.smithy.kotlin.runtime.net.Host
+import aws.smithy.kotlin.runtime.net.Scheme
+import io.ktor.network.tls.certificates.*
+import io.ktor.server.application.*
+import io.ktor.server.engine.*
+import io.ktor.server.netty.*
+import io.ktor.server.request.*
+import io.ktor.server.response.*
+import io.ktor.server.routing.*
+import kotlinx.coroutines.test.runTest
+import java.io.File
+import java.security.KeyStore
+import kotlin.test.Test
+import kotlin.test.assertEquals
+
+private const val CERTIFICATE_ALIAS = "certificate alias"
+private const val CERTIFICATE_PASSWORD = "certificate pass"
+private const val EXPECTED_REQUEST_BODY = "Do you accept self-signed certificates?"
+private const val EXPECTED_RESPONSE_BODY = "I *do* accept self-signed certificates!"
+private const val KEY_STORE_PASSWORD = "key store pass"
+private const val LOCALHOST = "127.0.0.1"
+private const val PATH = "/path/to/the/test"
+private const val PORT = 8443
+
+class CustomCertificateTest {
```

Comment:
**question**: Any reason we can't integrate this into our test suite against all engines (or at least work towards that) instead of making this okhttp specific?